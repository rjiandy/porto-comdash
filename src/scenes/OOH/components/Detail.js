// @flow

import React from 'react';
import {StyleSheet} from 'react-primitives';
import {View, Text} from '../../../general/components/coreUIComponents';
import {PlaceholderView} from '../../../general/components/UIComponents';
import PieChart from '../../../general/components/charts/PieChart';
import BarChart from '../../../general/components/charts/BarRechart';

import {THEME_COLOR, MEDIUM_GREY} from '../../../general/constants/colors';

import type {OOHDetailData} from '../type/OOH-type';

type Props = {
  data: OOHDetailData;
  selectedTerritory: string;
  selectedType: string;
};

export default function Detail(props: Props) {
  let {data, selectedTerritory, selectedType} = props;

  let filteredData = data.filter((datum) => datum.oohType === selectedType);

  let siteData = filteredData.filter((datum) => datum.itemType === 'SITE');

  let visualData = filteredData.filter((datum) => datum.itemType === 'VISUAL');

  let siteDataContent;
  if (siteData.length > 0) {
    siteDataContent = (
      <View style={[styles.chartItem, styles.borderContainer]}>
        <Text customStyle="title">By Site</Text>
        <PieChart
          data={siteData}
          xAxis="label"
          yAxis="measure"
          showTooltip
          showLegend
          showLabelValue
        />
      </View>
    );
  } else {
    siteDataContent = (
      <PlaceholderView
        text={`No site data available for territory ${selectedTerritory}, OOH type ${selectedType}`}
      />
    );
  }

  let visualDataContent;
  if (visualData.length > 0) {
    visualDataContent = (
      <View style={[styles.chartItem, styles.borderContainer]}>
        <Text customStyle="title">By Visual</Text>
        <BarChart
          data={visualData}
          xAxis="label"
          yAxis="measure"
          horizontal
          showTooltip
          showLabelValue
          showChartGrid
          extraProps={{
            bar: {
              fill: THEME_COLOR,
            },
          }}
        />
      </View>
    );
  } else {
    visualDataContent = (
      <PlaceholderView
        text={`No visual data available for territory ${selectedTerritory}, OOH type ${selectedType}`}
      />
    );
  }
  return (
    <View style={styles.root}>
      <View style={styles.flex}>
        {!selectedType ? (
          <PlaceholderView text="Please select OOH type first" />
        ) : (
          <View style={styles.chartContainer}>
            {siteDataContent}
            {visualDataContent}
          </View>
        )}
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  root: {
    flex: 1,
    flexDirection: 'row',
    paddingTop: 10,
  },
  flex: {
    flex: 1,
  },
  chartContainer: {
    flex: 1,
    flexDirection: 'row',
  },
  noDataAvailable: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    paddingHorizontal: 20,
  },
  noDataAvailableText: {
    textAlign: 'center',
  },
  chartItem: {
    flex: 1,
    marginHorizontal: 5,
    alignItems: 'center',
    justifyContent: 'center',
  },
  borderContainer: {
    borderRadius: 5,
    borderWidth: 1,
    borderColor: MEDIUM_GREY,
  },
});
