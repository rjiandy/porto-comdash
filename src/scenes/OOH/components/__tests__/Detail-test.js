// @flow

import React from 'react';
import {shallow} from 'enzyme';
import toJSON from 'enzyme-to-json';

import Detail from '../Detail';

it('should render Header component correctly', () => {
  let data = [
    {
      territory: 'Ambon',
      oohType: 'All',
      itemType: 'SITE',
      label: 'HMS',
      measure: 30,
    },
  ];
  let rendered = shallow(
    <Detail data={data} selectedTerritory="Ambon" selectedType="All" />
  );

  expect(toJSON(rendered)).toMatchSnapshot();
});
