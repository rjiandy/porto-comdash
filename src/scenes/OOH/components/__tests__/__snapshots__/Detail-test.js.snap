// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header component correctly 1`] = `
<View
  style={129}
>
  <View
    style={130}
  >
    <View
      style={131}
    >
      <View
        style={
          Array [
            134,
            135,
          ]
        }
      >
        <TextComponent
          customStyle="title"
        >
          By Site
        </TextComponent>
        <PieChartComponent
          data={
            Array [
              Object {
                "itemType": "SITE",
                "label": "HMS",
                "measure": 30,
                "oohType": "All",
                "territory": "Ambon",
              },
            ]
          }
          showLabelValue={true}
          showLegend={true}
          showTooltip={true}
          xAxis="label"
          yAxis="measure"
        />
      </View>
      <PlaceholderView
        text="No visual data available for territory Ambon, OOH type All"
      />
    </View>
  </View>
</View>
`;
